pkg/multitenant/hashing.go

```diff
@@ -0,0 +1,42 @@
+package multitenant
+
+import (
+	"context"
+
+	"github.com/kumahq/kuma/pkg/core/resources/store"
+)
+
+type Hashing interface {
+	GetOptionsFunc() store.GetOptionsFunc
+	ListOptionsFunc() store.ListOptionsFunc
```

Comment:
Shouldn't this just be `ResourceHashKey(ctx context.Context) string` and be a single function. You'd then remove the depency of this package on `resources/store` and in `resource/store/options` you'd define `type HashKeyFunc func(context.Context) string` and add to both `ListOptions` and `GetOptions` respectively `ListWithHashKey(HashKeyFund)` and `GetWithHashKey(HashKeyFunc)`. This leverages go's ducktyping to keep loosely coupled packages.